mantidproject/mantid
# Fix "next/previous run" co-adding run instead of replacing existing

Fixed a bug where the "load next/previous" would co-add the new run to the currently loaded run instead of replacing it in certain cases (where the file path contained the `-` character).

This is a small fix and it would be good to fix it for the release. However, the bug was present in previous releases, is not major and has not been seen by users, so it can be pushed back if necessary.

No test - the fix is in the GUI (untested) part of the MuonAnalysis interface.

**To test:**

Download the folder `\\olympic\babylon5\Scratch\TomPerkins\MUSR-test` and add it to Mantid's data path. These files are in the unit test data, but the important point is that the folder's name contains the character `-`.
- Open _Interfaces/Muon/Muon Analysis_
- Use the Browse button to load one of these files (do not load it by typing the run number in as this doesn't cause the bug to appear)
- Click 'next' or 'previous'
- Next or previous run should replace the currently loaded run, rather than being co-added to it.
  ![load](https://cloud.githubusercontent.com/assets/15363125/19072214/e8779ee0-8a2b-11e6-9668-760f9c9f2681.PNG)
- To test that co-adding runs still works when entering a range: type `15189-91` in the box and hit enter as below. Runs should be loaded and added together. If you click "next" it will extend the range to `00015189-00015192`.
  ![load2](https://cloud.githubusercontent.com/assets/15363125/19074916/805241d0-8a3a-11e6-9573-663ce649a72f.PNG)

Fixes #17688 

[Release notes](https://github.com/mantidproject/mantid/blob/425ca33ca649b481c2da4455edd5730e76c07f6b/docs/source/release/v3.9.0/muon.rst) updated. If pushed back to 3.9, the note needs to be moved to the new set of release notes.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
